examples/01-filter/image-fft-perlin-noise.py

```diff
@@ -0,0 +1,176 @@
+
+sampled_fft = sampled.fft()
+freq = np.fft.fftfreq(sampled.dimensions[0], sampled.spacing[0])
+max_freq = freq.max()
+
+# only show the first quadrant
+subset = sampled_fft.extract_subset((0, xdim // 2, 0, ydim // 2, 0, 0))
+
+
+###############################################################################
+# Plot the Frequency Domain
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+# Now, plot the noise in the frequency domain. Note how there is more high
+# frequency content in the x direction and this matches the frequencies given
+# to :func:`pyvista.perlin_noise <pyvista.core.common_data.perlin_noise>`.
+
+# scale by log to make the plot viewable
+subset['scalars'] = np.abs(subset.active_scalars.real)
+warped_subset = subset.warp_by_scalar(factor=0.0001)
+
+pl = pv.Plotter(lighting='three lights')
+pl.add_mesh(warped_subset, cmap='blues', show_scalar_bar=False)
+pl.show_bounds(
+    axes_ranges=(0, max_freq, 0, max_freq, 0, max_freq),
+    xlabel='X Frequency',
+    ylabel='Y Frequency',
+    zlabel='Amplitude',
+    show_zlabels=False,
+    color='k',
+    font_size=26,
+)
+pl.add_text('Frequency Domain of the Perlin Noise')
+pl.show()
+
+
+###############################################################################
+# Low Pass Filter
+# ~~~~~~~~~~~~~~~
+# Let's perform a low pass filter on the frequency content and then convert it
+# back into the space (pixel) domain by immediately applying a reverse FFT.
+#
+# When converting back, keep only the real content. The imaginary content has
+# no physical meaning in the physical domain. PyVista will drop the imaginary
+# content, but will warn you of it.
+#
+# As expected, we only see low frequency noise.
+
+low_pass = sampled_fft.low_pass(1.0, 1.0, 1.0).rfft()
+low_pass['scalars'] = low_pass.active_scalars.real
+warped_low_pass = low_pass.warp_by_scalar()
+warped_low_pass.plot(show_scalar_bar=False, text='Low Pass of the Perlin Noise', lighting=False)
+
+
+###############################################################################
+# High Pass Filter
+# ~~~~~~~~~~~~~~~~
+# This time, let's perform a high pass filter on the frequency content and then
+# convert it back into the space (pixel) domain by immediately applying a
+# reverse FFT.
+#
+# When converting back, keep only the real content. The imaginary content has no
+# physical meaning in the pixel domain.
+#
+# As expected, we only see the high frequency noise content as the low
+# frequency noise has been attenuated.
+
+high_pass = sampled_fft.high_pass(1.0, 1.0, 1.0).rfft()
+high_pass['scalars'] = high_pass.active_scalars.real
+warped_high_pass = high_pass.warp_by_scalar()
+warped_high_pass.plot(show_scalar_bar=False, text='High Pass of the Perlin Noise', lighting=False)
+
+
+###############################################################################
+# Sum Low and High Pass
+# ~~~~~~~~~~~~~~~~~~~~~
+# Show that the sum of the low and high passes equal the original noise.
```

Comment:
This is true because there were no imaginary parts. If there were and we discarded them, we would have discarded a component of the signal (perlin noise in this case).